peps/pep-0776.rst

```diff
@@ -0,0 +1,794 @@
+3. To identify minor features to be upstreamed from the Pyodide runtime into the
+   CPython Emscripten runtime
+
+The minor features identified here are all features that could be implemented
+without a PEP. We discuss more significant runtime features that we would like
+to implement but we defer decisions on those features to subsequent PEPs.
+
+Packaging Goals
+---------------
+
+1. To describe Pyodide's packaging tooling
+2. To describe Pyodide's wheel abi to a similar level of detail as the manylinux
+   PEPs
+3. To request that Pyodide wheels be allowed for upload to PyPI
+
+
+Emscripten Platform Information
+===============================
+
+Background on Emscripten
+------------------------
+
+`Emscripten
+<https://emscripten.org/docs/introducing_emscripten/about_emscripten.html>`__
+consists of a C and C++ compiler and linker based on LLVM, together with a
+runtime based on a mildly patched musl libc.
+
+Emscripten is a POSIX-based platform. It uses the WebAssembly binary format,
+specified here:
+
+* https://webassembly.github.io/spec/core/binary/index.html
+* https://github.com/WebAssembly/tool-conventions/blob/main/DynamicLinking.md
+
+The ``emcc`` compiler is a wrapper around ``clang``. The ``emcc`` linker is a
+wrapper around ``wasm-ld`` (also part of the LLVM toolchain).
+
+Emscripten support for portable C/C++ code source compatibility with Linux is
+fairly comprehensive, with certain expected exceptions to be spelled out.
+CPython already supports compilation to Emscripten, and it only requires a very
+modest number of modifications to the normal Linux target.
+
+POSIX Compliance
+----------------
+
+Emscripten is a POSIX platform. However, there are POSIX APIs that exist but
+always fail when called and POSIX APIs that don't exist at all. In particular,
+there are problems with networking APIs and blocking I/O, and there is no
+support for ``fork()``. See `Emscripten Portability Guidelines
+<https://emscripten.org/docs/porting/guidelines/portability_guidelines.html>`__.
+
+Emscripten executables can be linked with threading support, but it comes
+with several limitations:
+
+* Enabling threading requires websites to be served with special security headers
+  that indicate acceptance of the possibility of `Spectre <https://en.wikipedia.org/wiki/Spectre_(security_vulnerability)>`__-style information
+  leakage. These headers are a usability hazard for users who are not intimately
+  familiar with the web platform.
+
+* If an executable is linked with both threading and a dynamic loader, Emscripten
+  prints a warning that using dynamic loading and pthreads together is
+  experimental. It may cause performance problems or crashes. These problems may
+  require WebAssembly standards work to resolve.
+
+Because of these limitations, Pyodide standardizes a no-pthreads build of
+Python. We will start with support for no-pthreads builds. If there is
+sufficient demand, a pthreads build with no dynamic loader could be added later.
```

Comment:
The second sentence seems to repeat the first
```suggestion
Because of these limitations, Pyodide standardizes a no-pthreads build of
Python. If there is sufficient demand, a pthreads build with no dynamic loader
could be added later.
```